src/posix/platform/netif.cpp

```diff
@@ -152,6 +152,40 @@ char         gNetifName[IFNAMSIZ];
 
 #if OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
 
+namespace {
+
+/**
+ * This utility class converts binary IPv6 address to text format.
+ *
+ */
+class Ip6AddressString
+{
+public:
+    /**
+     * The constructor of this converter.
+     *
+     * @param[in]   aAddress    A pointer to a buffer holding an IPv6 address.
+     *
+     */
+    Ip6AddressString(const void *aAddress)
```

Comment:
```suggestion
    Ip6AddressString(const otIp6Address *aAddress)
```

Reply:
This is for all IPv6 address structures or objects following the inet_ntop API.

Reply:
It is. But we only use it for `const otIp6Address *aAddress`? Use `const otIp6Address *aAddress` for better type checking?